web-api/terraform/modules/batch/docker-image/src/docketEntryBatchDownloader.ts

```diff
@@ -0,0 +1,263 @@
+      Key: outputZipName,
+    },
+  });
+
+  const writable = new Writable({
+    write(chunk, encoding, callback) {
+      callback();
+    },
+  });
+
+  passThrough.pipe(writable);
+
+  // Start the upload process
+  const uploadPromise = upload.done();
+
+  const zip = new Zip((err, data, final) => {
+    if (err) {
+      console.log('Error creating zip stream');
+      throw err;
+    }
+
+    passThrough.write(data);
+
+    if (final) {
+      passThrough.end();
+    }
+  });
+
+  for (let index = 0; index < documents.length; index++) {
+    const { filePathInZip, key, useTempBucket } = documents[index];
+    console.log(`Downloding document (${key})`);
+
+    const response = await storageClient.getObject({
+      Bucket: useTempBucket ? TEMP_S3_BUCKET : S3_BUCKET,
+      Key: key,
+    });
+
+    if (!response.Body) {
+      throw new Error(`Unable to get document (${key}) from persistence.`);
+    }
+
+    // Transform s3 getobject into a stream of data that can be piped into the zip processor
+    const bodyStream: ReadableStream<Uint8Array> =
+      response.Body.transformToWebStream();
+    const reader = bodyStream.getReader();
+    const compressedPdfStream = new AsyncZipDeflate(filePathInZip);
+    zip.add(compressedPdfStream);
+
+    let continueReading = true;
+    while (continueReading) {
+      const unzippedChunk = await reader.read();
+      const nextChunk = unzippedChunk.value || new Uint8Array();
+      continueReading = !unzippedChunk.done;
+      compressedPdfStream.push(nextChunk, unzippedChunk.done);
+      while (passThrough.readableLength > 1024 * 1024 * 10) {
+        // Wait for the buffer to be drained, before downloading more files
+        await new Promise(resolve => setTimeout(resolve, 10));
+      }
+    }
+    reader.releaseLock();
+
+    console.log(`Download Complete (${key})`);
+    const WS_MESSAGE = new PostToConnectionCommand({
+      ConnectionId: connectionId,
+      Data: JSON.stringify({
+        action: 'batch_download_progress',
+        filesCompleted: index + 1,
+        totalFiles: documents.length,
+      }),
+    });
+    await wsClient.send(WS_MESSAGE);
+  }
+
+  zip.end();
+  await uploadPromise;
+}
+
+export async function app({
+  connectionId,
+  documentsReference,
+  s3Client,
+  wsClient,
+  zipName,
+}: DocketEntriesZipperParameter) {
+  console.log('LOADING DOCUMENTS TO DOWNLOAD FROM S3');
+
+  const documents: DocketEntryDownloadInfo[] = await getDocketEntriesFromS3(
+    documentsReference,
+    s3Client,
+  );
+  console.log('ZIPPING DOCUMENTS');
+
+  const guid = Date.now();
+  const UNIQUE_ZIP_NAME = `${guid}/${zipName}`;
+
+  await zipDocuments({
+    connectionId,
+    documents,
+    outputZipName: UNIQUE_ZIP_NAME,
+    wsClient,
+  });
+
+  console.log('Fetching Signed URL');
+  const command = new GetObjectCommand({
+    Bucket: ZIP_TEMP_S3_BUCKET,
+    Key: UNIQUE_ZIP_NAME,
+  });
+
+  const url = await getSignedUrl(zipStorageClient, command, { expiresIn: 120 });
```

Comment:
Am I misreading, or does this expire after two minutes? If so, this seems too short. (The user might leave their laptop as the download starts and come back after a few minutes, but the link won't work.)

Reply:
The link is generated and immediately sent to front end via web socket message and it starts downloading (all automated).

Reply:
What if the pop-up is blocked? In my case, I had to manually reclick the link via the popup icon:

<img width="302" alt="Screenshot 2024-10-29 at 11 00 01 AM" src="https://github.com/user-attachments/assets/6b2eb8f6-98bf-4385-a4bf-9d689c1cbb6e">

Reply:
I would support a longer expiration as well.